server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4118,7 +4134,11 @@ public void close() throws IOException {
                 }
             };
             IOUtils.close(currentEngineReference.getAndSet(readOnlyEngine));
-            newEngineReference.set(engineFactory.newReadWriteEngine(newEngineConfig(replicationTracker)));
+            EngineConfig newEngineConfig = newEngineConfig(replicationTracker);
+            if (indexSettings.isRemoteStoreEnabled()) {
+                syncSegmentsFromRemoteSegmentStore(false);
+            }
+            newEngineReference.set(engineFactory.newReadWriteEngine(newEngineConfig));
```

Comment:
why did split setting `newEngineReference` into two steps ? can we do it like 

```
            if (indexSettings.isRemoteStoreEnabled()) {
                syncSegmentsFromRemoteSegmentStore(false);
            }
            newEngineReference.set(engineFactory.newReadWriteEngine(newEngineConfig(replicationTracker)));

```

Reply:
I am not sure. :) I will get it changed.